### PR TITLE
Add JSON tag for ErrorResponse

### PR DESCRIPTION
**Pull request by migueleliasweb**
Fixes: #2640.

### CODE REVIEW

**Comment by gmlewis** [comment]
Why is this even needed?
What problem is it solving?

File: github/github.go
Comment (gmlewis) [inline github/github.go:844]: I've never seen a bare "omitempty" before, so can we please name it as "response"?

```suggestion
	Response *http.Response `json:"response,omitempty"` // HTTP response that caused this error
```

Comment (migueleliasweb) [reply github/github.go:844]: I'm fine adding the name of the field. But I wonder in this case if it might needs to be `Response` to maintain compatibility with the current code.

Comment (gmlewis) [reply github/github.go:844]: You are saying it should be "Response" with a capital-R ?
When you say "compatibility with the current code", do you mean the "go-github" repo?
If so, I'm not sure why we would need to capitalize the R, and I would think the lower-case version should be fine.

Alternatively, another approach might be to simply use `json:"-"` and always leave it out... but I'll let you decide if that is the best option for compatibility with your mocking package.

Comment (migueleliasweb) [reply github/github.go:844]: Hi Glenn, I actually liked the idea of going with `json:"-"`. Let me test a few things and I will come back to you.

Thanks for the swift responses!

Comment (migueleliasweb) [reply github/github.go:844]: Hi @gmlewis , it seems the `json:"-"`  should work just fine. I will update the PR soon.

Comment (migueleliasweb) [reply github/github.go]: done :)

**Review by gmlewis — APPROVED** [review]
Thank you, @migueleliasweb !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by migueleliasweb** [comment]
It's always a pleasure, @gmlewis  :smiley:  . Is there someone I could ping here to get this going?

**Comment by gmlewis** [comment]
@Parker77 - do you have time for an LGTM+Approval, please?

**Review by Parker77 — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @Parker77!
Merging.

**Comment by migueleliasweb** [comment]
Btw, merging this fixed the default error formatting issue on `go-github-mock` 😃 . Thanks all!

**Pull request merged**
